### ISSUE TITLE
Set-Date did not work

### ISSUE THREAD

**Issue by chucklu** [PowerShell/PowerShell#5163]
Steps to reproduce
------------------

```powershell
Set-Date -Date (Get-Date).AddDays(3)
```
I  find this command on [official document](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/set-date?view=powershell-6)

Expected behavior
-----------------

```none
should change the datetime on my computer
```

Actual behavior
---------------

```none
Set-Date : The parameter is incorrect
At line:1 char:1
+ Set-Date -Date (Get-Date).AddDays(3)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Set-Date], Win32Exception
    + FullyQualifiedErrorId : System.ComponentModel.Win32Exception,Microsoft.PowerShell.Commands.SetDateCommand
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.6
OS                             Microsoft Windows 10.0.10586
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

I also checked the issue list, and find another issue about [set-date test](https://github.com/PowerShell/PowerShell/issues/3099). I am not sure if they are the same issue.

**Comment by SteveL-MSFT** [comment]
This works on Ubuntu, but fails on Windows 10 for me

**Comment by stknohg** [comment]
I reproduced this problem on Windows 10.

`Microsoft.PowerShell.Commands.SetDateCommand.SystemTime` should be `struct`.

https://github.com/PowerShell/PowerShell/blob/2f818615bed15141c062dd185f659ed110d9c6ba/src/Microsoft.PowerShell.Commands.Utility/commands/utility/SetDateCommand.cs#L117-L134

**Comment by MaximoTrinidad** [comment]
@chucklu, Please make sure to use the latest version of PSCore6 ( 6.0.0 GA) as Beta is gone.
:)

Confirmed! 
Works in Ubuntu but not Windows 10.

![ubuntusetdate_01_2018-01-24_9-18-40](https://user-images.githubusercontent.com/19892435/35336874-b6a55658-00e7-11e8-8ddc-0d2c0659d474.jpg)
![windowssetdate_01_2018-01-24_9-14-17](https://user-images.githubusercontent.com/19892435/35336875-b6b408ec-00e7-11e8-979f-2175aa7dc085.jpg)

**Comment by chucklu** [comment]
@MaximoTrinidad  I  have checked the release list, the latest version is released five days ago with sha-1 2f81861.
I will waiting for next release version, thanks.
